aten/src/ATen/native/cpu/DepthwiseConvKernel.cpp

```diff
@@ -282,12 +284,13 @@ Tensor _convolution_depthwise3x3_winograd(
                       bias_potentially_undefined :
                       at::zeros({kernel_sizes[0]}, input.options());
 
-  at::parallel_for(0, args.batch * groups, 0, [&](int64_t start, int64_t end) {
+  at::parallel_for(0, args.batch * args.out_channels, 0, [&](int64_t start, int64_t end) {
     for (int64_t k = start; k < end; ++k) {
-      const int64_t g = k % groups;
+      const int64_t g = k % args.out_channels;
+      const int64_t i = g % groups;
```

Comment:
From writing it on paper, this is what I get:
```suggestion
      const int64_t i = k / (args.out_channels / groups);
```
Namely, how many groups of outputs did you go through. That will correspond to the index after linearizing the batch size.

I'm really bad at modulo computation but the two are not equivalent right?

Reply:
You are right, two are not equivalent, but I think both formulas are wrong. Take `nn.Conv2d(1, 32, (3, 3))` for example. groups for such convolution would be 1, out_channels 32, but i should always be 0 if batch == 1, which is the case for my formula, but not for yours.

Reply:
Isn't it? k is in range `[0, out_channels[` for batch==1. So `k / out_channels` will always be 0

Reply:
You are right, it is a correct formula (using module would pick interleave input channels in the wrong order)